<?xml version="1.0"?>
<!-- Launches the fence detector node-->
<!-- vim: set fenc=utf-8 et sts=4 ft=xml : -->
<launch>

    <!-- Fence Nodelet Host process -->
    <node
        machine="navigate"
        pkg="nodelet"
        type="nodelet"
        name="manager"
        args="manager"
        ns="fence"
        output="log"
        />

    <!-- Compute Stereo Point Cloud -->
    <node
        machine="navigate"
        pkg="nodelet"
        type="nodelet"
        name="stereo_image_points"
        args="load stereo_image_proc/point_cloud2 /fence/manager"
        ns="fence"
        output="log"
        >
        <param name="approximate_sync" type="bool" value="true"/>
        <remap from="left/image_rect_color" to="fence_img"/>
        <remap from="left/camera_info" to="/cameras/navigation/left/camera_info"/>
        <remap from="right/camera_info" to="/cameras/navigation/right/camera_info"/>
        <remap from="disparity" to="fence_disp"/>
    </node>

    <!-- Fence Detector -->
    <node
        machine="navigate"
        pkg="fence_detection"
        type="fence_detection.py"
        name="fence_detection"
        ns="fence"
        output="log">
        <param name="min_h" type="int" value="5"/>
        <param name="max_h" type="int" value="15"/>
        <param name="min_s" type="int" value="155"/>
        <param name="max_s" type="int" value="180"/>
        <remap from "input_image" to="/cameras/navigation/center/left/image_rect_color"/>
        <remap from "input_disparity" to="/cameras/navigation/center/disparity"/>
    </node>

</launch>
